docs/design/02-topologies.md

```diff
@@ -0,0 +1,288 @@
+
+# Overview
+
+- Will have new `metal configure` command group for configuring values to be
+  used within config files for all nodes, a group of nodes, or an individual
+  node:
+  - `metal configure all`
+  - `metal configure group GROUP`
+  - `metal configure node NODE`
+
+- These commands may also perform other behaviour specific to the type of
+  resource being configured.
+
+- Note that these commands will be just one interface for configuring the
+  'topology' of a cluster; it should be possible for alternative interfaces,
+  such as a future web interface or manual editing, to modify the same files,
+  and the loading of these files for use when templating will be independent of
+  the `configure` commands.
+
+- There will be a `configure.yaml` file in the repo root, specifying
+  configuration questions to be used within these commands or alternative
+  interfaces.
+
+- The `configure` commands will save the entered answers to a file in
+  `/var/lib/metalware/config`, specific to the resource being configured.
+
+- Repo configs will be able to use a new object in the magic `alces` namespace,
+  `alces.config.$question_identifier`, where `$question_identifier` is the
```

Comment:
Yeah I agree, added something briefly about that here: https://github.com/alces-software/metalware/pull/82/files/395b7823017437099db9a3b9fb7f60fdb11b9512#diff-7633f31d52d6afaac7f6adf486a7e485R286. Not sure what the best terms to use might be, any ideas?

@ste78 would you describe the 'topology' of the cluster as being what we currently call the 'config' in the repo, or the new planned config files for `all`, `groups` and `nodes` - or the combination of both of these? Wondering if one of these should be renamed to `topology` for clarity.